riscv/insns/csrrwi.h

```diff
@@ -1,5 +1,9 @@
 int csr = validate_csr(insn.csr(), true);
-reg_t old = p->get_csr(csr, insn, true);
+bool read = insn.rd() != 0;
+reg_t old = 0;
+if (read) {
+  old = p->get_csr(csr, insn, true);
+}
```

Comment:
A more compact version of this would be something like

    // Avoid actioning read side effects when rd is 0
    reg_t old = insn.rd() != 0 ? p->get_csr(csr, insn, true) : 0;

Reply:
Got it.